tasks/clamav-scan.yaml

```diff
@@ -0,0 +1,81 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: clamav-scan
+spec:
+  params:
+    - name: image-digest
+      description: Image digest to scan
+    - name: image-url
+      description: Url to image
+  steps:
+    - name: extract-and-scan-image
+      image: quay.io/redhat-appstudio/hacbs-test:v1.0.1
+      script: |
+        # copy virus database from sidecar
+        cp -r /tmp/clamdb/* /var/lib/clamav
+
+        # strip new-line escape symbol from parameter and save it to variable
+        imageanddigest=$(echo '$(params.image-url)'@'$(params.image-digest)' | tr -d '\n')
+
+        mkdir -p /tmp/extracted-image && cd /tmp/extracted-image
+        oc image extract -a /workspace/registry-auth/.dockerconfigjson $imageanddigest
+        clamscan /tmp/extracted-image -ri --max-scansize=250M | tee /tekton/home/clamscan-result.log
+        echo "Executed-on: Scan was executed on version - $(clamscan --version)" | tee -a /tekton/home/clamscan-result.log
+      volumeMounts:
+        - mountPath: /tmp/clamdb
+          name: dbfolder
+    - name: modify-clam-output-to-json
+      image: python
+      script: |
+        #!/usr/bin/env python
+        import json
+
+        with open("/tekton/home/clamscan-result.log", "r") as file:
+            clam_result_str = file.read()
+
+        def clam_result_str_to_json(clam_result_str):
+
+            clam_result_list = clam_result_str.split("\n")
+            clam_result_list.remove('')
+
+            results_marker = \
+                clam_result_list.index("----------- SCAN SUMMARY -----------")
+
+            hit_list = clam_result_list[:results_marker]
+            summary_list = clam_result_list[(results_marker + 1):]
+
+            r_dict = { "hits": hit_list }
+            for item in summary_list:
+                # in case of blank lines
+                if not item:
+                    continue
+                split_index = [c == ':' for c in item].index(True)
+                key = item[:split_index].lower()
+                key = key.replace(" ", "_")
+                value = item[(split_index + 1):].strip(" ")
+                r_dict[key] = value
+            print(json.dumps(r_dict))
+            with open('/tekton/home/clamscan-result.json', 'w') as f:
+              print(json.dumps(r_dict), file=f)
+
+        def main():
+            clam_result_str_to_json(clam_result_str)
```

Comment:
I was wondering if it would make sense to convert the date values to ISO format? I mean it depends on who/what will be reading the output, but since it's already processed somehow here, I think it could be helpful to process it a bit more so the possible following steps will have it easier to process. What do you think?

Reply:
I think this won't hurt anyone if the date is in isoformat. Making adjustments already. I changed the ones in json output
`start_date` & `end_date`. The verbose one at the very end would need more complicated handling, which could be done later if requested. Does this change meets the requirements?

Reply:
Yes thank you!